src/ethrpc.ts

```diff
@@ -28,3 +29,25 @@ export const transactionListener = (provider: any, tx: string, userCallback: any
       .catch((err: Error) => userCallback({ error: err.message }))
   , 2000)
 }
+
+export enum PROVIDERS {
+  METAMASK = 'METAMASK',
+  NIFTY = 'NIFTY',
+  WALLET_CONNECT = 'WALLET_CONNECT'
+}
+
+/**
+ * Return the Provider Name. Used to detect DataVault features
+ * @param provider web3Provider
+ */
+export const getProviderName = (provider: any) => {
+  if (provider.isNiftyWallet) {
+    return PROVIDERS.NIFTY
+  }
+  if (provider.isMetaMask) {
+    return PROVIDERS.METAMASK
+  }
+  if (provider.wc) {
+    return provider.WALLET_CONNECT
```

Comment:
@jessgusclark 

`return provider.WALLET_CONNECT`
     
wouldn't it be `return PROVIDERS.WALLET_CONNECT` ?

Reply:
@miltonwg Yes, you are correct, thanks for catching this.